server/storage/database_test.go

```diff
@@ -2,290 +2,326 @@ package storage
-		Data:    []byte("2"),
-	}
-	// start transation
-	sqlmock.ExpectBegin()
-
-	sqlmock.ExpectQuery("SELECT count\\(\\*\\) FROM `tuf_files` WHERE `gun`=\\? AND `role`=\\? AND `version`>=\\?\\;").WithArgs(
-		"testGUN",
-		update1.Role,
-		update1.Version,
-	).WillReturnRows(sqlmock.RowsFromCSVString([]string{"count(*)"}, "0"))
-	sqlmock.ExpectExec("INSERT INTO `tuf_files` \\(`gun`, `role`, `version`, `data`\\) VALUES \\(\\?,\\?,\\?,\\?\\);").WithArgs(
-		"testGUN",
-		update1.Role,
-		update1.Version,
-		update1.Data,
-	).WillReturnResult(sqlmock.NewResult(0, 1))
-
-	sqlmock.ExpectQuery("SELECT count\\(\\*\\) FROM `tuf_files` WHERE `gun`=\\? AND `role`=\\? AND `version`>=\\?\\;").WithArgs(
-		"testGUN",
-		update2.Role,
-		update2.Version,
-	).WillReturnRows(sqlmock.RowsFromCSVString([]string{"count(*)"}, "0"))
-	sqlmock.ExpectExec("INSERT INTO `tuf_files` \\(`gun`, `role`, `version`, `data`\\) VALUES \\(\\?,\\?,\\?,\\?\\);").WithArgs(
-		"testGUN",
-		update2.Role,
-		update2.Version,
-		update2.Data,
-	).WillReturnResult(sqlmock.NewResult(0, 1))
-
-	// expect commit
-	sqlmock.ExpectCommit()
-
-	err = s.UpdateMany(
-		"testGUN",
-		[]MetaUpdate{update1, update2},
-	)
-	assert.Nil(t, err, "UpdateMany errored unexpectedly: %v", err)
-
-	// There's a bug in the mock lib
-	//err = db.Close()
-	//assert.Nil(t, err, "Expectation not met: %v", err)
 }
 
-func TestMySQLUpdateManyRollback(t *testing.T) {
-	db, err := sqlmock.New()
-	assert.Nil(t, err, "Could not initialize mock DB")
-	s := NewMySQLStorage(db)
-	update := MetaUpdate{
+func SampleUpdate(version int) MetaUpdate {
+	return MetaUpdate{
 		Role:    "root",
-		Version: 0,
+		Version: version,
 		Data:    []byte("1"),
 	}
-	execError := mysql.MySQLError{}
-	// start transation
-	sqlmock.ExpectBegin()
-
-	sqlmock.ExpectQuery("SELECT count\\(\\*\\) FROM `tuf_files` WHERE `gun`=\\? AND `role`=\\? AND `version`>=\\?\\;").WithArgs(
-		"testGUN",
-		update.Role,
-		update.Version,
-	).WillReturnRows(sqlmock.RowsFromCSVString([]string{"count(*)"}, "0"))
-	// insert first update
-	sqlmock.ExpectExec("INSERT INTO `tuf_files` \\(`gun`, `role`, `version`, `data`\\) VALUES \\(\\?,\\?,\\?,\\?\\);").WithArgs(
-		"testGUN",
-		update.Role,
-		update.Version,
-		update.Data,
-	).WillReturnError(&execError)
-
-	// expect commit
-	sqlmock.ExpectRollback()
-
-	err = s.UpdateMany(
-		"testGUN",
-		[]MetaUpdate{update},
-	)
-	assert.IsType(t, &execError, err, "UpdateMany returned wrong error type")
-
-	err = db.Close()
-	assert.Nil(t, err, "Expectation not met: %v", err)
 }
 
-func TestMySQLUpdateManyDuplicate(t *testing.T) {
-	db, err := sqlmock.New()
-	assert.Nil(t, err, "Could not initialize mock DB")
-	s := NewMySQLStorage(db)
-	update := MetaUpdate{
-		Role:    "root",
-		Version: 0,
-		Data:    []byte("1"),
+// SetUpSQLite creates a sqlite database for testing
+func SetUpSQLite(t *testing.T) (*gorm.DB, *MySQLStorage) {
+	tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+	defer os.RemoveAll(tempBaseDir)
```

Comment:
I originally thought so too, but since the DB seemed fine during the tests, I got confused and thought maybe I misunderstood how this worked.  Thanks for noticing!  :)